src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -170,77 +170,92 @@ public int GetLocalHostPort() {
     }
 
     public String GetLocalHostAddress() {
-        String ret = "";
-        if(localHostSocket != null && localHostSocket.getInetAddress() != null){
-            ret = localHostSocket.getInetAddress().toString();
+        if(localHostSocket == null){
```

Comment:
From a style perspective I would have done something like
```Java
return localHostSocket == null || localHostSocket.getInetAddress() == null ? null : localHostSocket.getInetAddress().toString();
```

Although to be fair calling a function twice (in this case getInetAddress) isn't really best practice but it's cheap enough that it's probably not worth worrying about.

Oh and I would suggest renaming the function as GetLocalHostAddressAsString.